packages/frontend/core/src/mobile/components/doc-card/index.tsx

```diff
@@ -64,12 +72,40 @@ export const DocCard = forwardRef<HTMLAnchorElement, DocCardProps>(
       return { height: `${rows * 18}px` };
     }, [autoHeightById, meta.id]);
 
+    const [visible, setVisible] = useState(false);
+
+    useEffect(() => {
+      const observer = new IntersectionObserver(entries => {
```

Comment:
Better to use shared Observer for those cards since there may be thousands of cards. I've tested locally and there is indeed a significant difference, mainly in terms of CPU consumption and JS heap size.